### DISCUSSION TITLE
Vias not generated properly

### DISCUSSION THREAD

**Discussion by gkamendje** [The-OpenROAD-Project/OpenROAD-flow-scripts#1577] [Q&A]
I am seeing a very strange behavior on the Vias placed in the design. All Vias type are affected. The Vias are rectangle instead of being square. 
In OpenRoad they look like this:
![Via_issue](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/56942214/28831da9-2719-4aaa-b158-6ec69b01ace0)

I was Instead expecting the vias to look like this:

![ExpectedVia](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/56942214/63032133-d509-423b-ac94-4cb1869c8cc6)

Here a more closer look at the generated vias

![WrongVias](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/56942214/eeac24f8-3369-4a1f-bc64-29c6b3946e76)

Not sure whether this is a tool issue or an issue with my platform setup.  
This behavior leads to minimum spacing, minimum density and minimum area DRC violations. 

Like here on M2

![ViaDRC1](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/56942214/27fe576e-dc95-44db-8348-aa42629ce226)

And here on M4

![ViaDRC2](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/56942214/40670652-f6ca-474e-b0c5-3263a741bbf2)

Here is a testcase to reproduce the issue
https://drive.google.com/file/d/118weCrMFzEWYQUv8CcZXcbVWtMFq8Rfy/view?usp=share_link

**Comment by maliberty** [comment]
This is better filed as an issue than a discussion.

**Reply by gkamendje** [reply]
I was not sure that this is tool issue. I was under the impression that it could be a platform setup issue. Is there a way to promote it to an issue? If not I will manually file it as an issue

**Comment by maliberty** [comment]
are these vias in the power grid?

**Reply by gkamendje** [reply]
No the vias are on the signals

**Comment by gkamendje** [comment]
Closing this as it was moved to issue #1578

**Reply by rovinski** [reply]
@gkamendje In my mind, the best way to distinguish if you should file an issue or a discussion is "do I have a log file related to this problem?". If yes, you should file an issue because there is a template which provides us with many necessary details to understand the environment and context of the problem. Without these, developers end up playing a guessing game and it doesn't become an efficient use of anyone's time.

If your problem doesn't have a log file, it is likely better as a discussion. This could be things like discussing high-level or abstract concepts, asking questions about OpenROAD's algorithms or implementation, asking for feedback from developers and/or the community, asking about general strategies, and so on.
